module/Entra/customizations/Get-EntraUser.ps1

```diff
@@ -45,16 +46,14 @@
         if($null -ne $PSBoundParameters["ObjectId"])
         {
             $UserId = $PSBoundParameters["ObjectId"]
-            if ([Guid]::TryParse($UserId, [ref]([Guid]::Empty))) {
+            if ($UserId -match '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'){
+                $f = '$' + 'Filter'
+                $Filter = "UserPrincipalName eq '$UserId'"
```

Comment:
The endpoint support this call
`GET /users/{id | userPrincipalName}`
See https://learn.microsoft.com/en-us/graph/api/user-get?view=graph-rest-1.0&tabs=http#http-request

Is there an easy way to avoid using `Filter` query?

Reply:
Yes, it supports `GET /users/{id | userPrincipalName}`. 
However, this API endpoint does not return all values by default. To retrieve all properties, we use the `$select` query. Due to this, using `userPrincipalName` in the endpoint throws the following error.

![image](https://github.com/user-attachments/assets/214706f9-541d-48c9-910f-43d50bd93ec7)